### PR TITLE
captain's spare must be unlocked to get all access

### PR DESCRIPTION
**Pull request by d3athrow**
otherwise it has maint and bridge access only.

Durr

:cl:
 * experiment: Captain's Spare requires a fingerprint of a head of staff to unlock full access, otherwise it is limited to Bridge and Maintenance only. If there are no heads of staff in the manifest the spare will unlock regardless.

### CODE REVIEW

**Comment by Kharkov72** [comment]
I think handstealing is neat, but I worry about lowpop, since they can quite easily have nobody at all to unlock the spare. Even in highpop there are sometimes no heads.

**Comment by d3athrow** [comment]
@Kharkov72 I can try and make it so it unlocks regardless if there are no heads of staff on the manifest.

File: code/game/objects/items/weapons/cards_ids.dm [hunk 1]
```diff
@@ -583,12 +583,53 @@
...
-	access = J.get_access()
+	locked_access = J.get_access()
+	access = list(access_maint_tunnels, access_heads)
 	..()
 
+/obj/item/weapon/card/id/captains_spare/proc/toggle_access_lock(mob/user, var/datum/data/record/R, var/own_hand = TRUE)
+	if(!R)
+		to_chat(user, "<span class = 'warning'>\The [src] makes a soft buzzing noise as [own_hand ? "your fingerprint" : "the severed hand's fingerprint"] is rejected</span>")
+		return
+	if(R.fields["rank"] == "Captain" || R.fields["rank"] == "Head of Security" || R.fields["rank"] == "Research Director" || R.fields["rank"] == "Chief Medical Officer" || R.fields["rank"] == "Head of Personnel" || R.fields["rank"] == "Chief Engineer")
```
Comment (N3X15) [inline code/game/objects/items/weapons/cards_ids.dm]: There should be some const or define for these somewhere.

May also want to change this to a `ALLOWED_RANKS.Find(R.fields["rank"])` check, where `ALLOWED_RANKS` is a `list()` of ranks.

File: code/game/objects/items/weapons/cards_ids.dm [hunk 2]
```diff
@@ -583,12 +583,53 @@
...
+		if(fLocked)
+			to_chat(user, "<span class = 'warning'>\The [src] makes a soft beep as [own_hand ? "your fingerprint" : "the severed hand's fingerprint"] unlocks the ID's full access.</span>")
+			fLocked = FALSE
```
Comment (N3X15) [inline code/game/objects/items/weapons/cards_ids.dm]: You can move the `fLocked` after the `if()` and change it to `fLocked = !fLocked`

**Comment by Dacendeth** [comment]
Does the impact the wizard spell summon spares?

**Review by N3X15 — APPROVED** [review]
Beyond a few minor quibbles, looks OK.

File: code/game/objects/items/weapons/cards_ids.dm (hunk 1 above)
Comment (ShiftyRail) [reply code/game/objects/items/weapons/cards_ids.dm]: `R.fields in command_positions`

File: code/game/objects/items/weapons/cards_ids.dm (hunk 2 above)
Comment (d3athrow) [reply code/game/objects/items/weapons/cards_ids.dm]: What do you mean move it after the if()?

Comment (N3X15) [reply code/game/objects/items/weapons/cards_ids.dm]: After the `if(fLocked)` block, I mean.  Not critical.

File: code/game/objects/items/weapons/cards_ids.dm [hunk 3]
```diff
@@ -583,12 +583,67 @@
 	item_state = "gold_id"
 	registered_name = "Captain"
 	assignment = "Captain"
+	var/list/locked_access
+	var/fLocked = TRUE
 
+/obj/item/weapon/card/id/captains_spare/summoned
```
Comment (Kurfursten) [inline code/game/objects/items/weapons/cards_ids.dm:594]: you don't actually have to put a separate definition here if you're not changing any vars, although you might add a cheeky description like `This one looks extra magical.`

**Comment by Kurfursten** [comment]
I think what you did with the hands and arms is really cute. Very classic SS13 Easter egg.

**Comment by robosperg** [comment]
would be nice if description reflected changes, and showed if it's unlocked or not

**Comment by utilitas-** [comment]
\>yet another retarded spare rushing code "fix"
\>except this time it gets widely accepted because the chicken made it
hmmm
I'd appreciate it if you didn't pollute the codebase with shit changes meant to fix a single server's shit admin issues so non-/vg/ servers can actually use it without reverting every third commit.

**Comment by boy2mantwicethefam** [comment]
But your ilk polluted the thread too, we can't all have nice things

**Comment by d3athrow** [comment]
@utilitas- Don't you have some erp to do, degenerate?

**Comment by utilitas-** [comment]
@d3athrow Don't you have a server to run, degenerate? Get back to your icecream handouts.

**Comment by d3athrow** [comment]
Don't you dare talk shit about my ice cream handouts

**Comment by MisterStaub** [comment]
@utilitas- sucks to suck

**Comment by ghost** [comment]
Interesting to see where this goes.

**Comment by ShiftyRail** [comment]
I think this is completely unnecessary baby-proofing. While I agree that spare rushing can be an issue, the real problem (for me...) was always the shitters who loot the cap laser and the lasgun and use it to play town sheriff and self-antagonise because "now security is after me, so this means I can wall them off in maint/try to slip them into space"/whatever else.

**Comment by Shadowmech88** [comment]
While I personally feel like spare "rushing" should fall under the purview of admins a la "don't be a dick," this is definitely the most agreeable of all the ways to fix it via code that I've seen proposed.

**Comment by boy2mantwicethefam** [comment]
@ShiftyRail Literally never happens

**Comment by Diavill** [comment]
the new meta is raping a head of staff, stealing their DNA and then rushing the golden ticket, can't wait

**Comment by utilitas-** [comment]
can't wait for heads of staff become even more like items than they already are
new speedrun tactic involves capping the RD the moment he walks out of science then falco stomping into the bridge with his corpse to unlock the spare by 3:00
in fact, this incentivizes rushing the spare _when there is a captain_ because then you can _MUH ESCALATION_ yourself way to killing/tabling the captain and taking his ID instead of the spare (but it's all okay because you revive him afterwards :) )
/vg/station: the pioneer in finding solutions to non-issues instead of solving the root of the problem (hint hint hint: it's the players)

**Comment by kilozombie** [comment]
@utilitas- what the fuck is this fantasy realm you're describing

excuse me, because i don't play much at all nowadays, but since fucking when have nonantags been able to kill a head of staff and use their ID to coast into the spare, EVER

if people are actively killing the captain because of muh access, it is infinitely easier for them to be banned than if they're sparerushing

**Comment by utilitas-** [comment]
wake up, old man, this ain't your space station of yesteryear
this is a doggy dog station and there ain't one bitch who'll gnaw on your bone if you don't hunt yourself some CMOs nowadays

**Comment by Probe1** [comment]
Testmerge

**Pull request merged**